ui/src/components/router/component-auto-binder.tsx

```diff
@@ -0,0 +1,29 @@
+type BindableProps<TPropName extends string, TValue> = {
+	[P in TPropName]?: TValue;
+};
+
+export type ComponentAutoBinder<TPropName extends string, TValue> = <
+	TProps extends BindableProps<TPropName, TValue>,
+>(
+	Component: React.ComponentType<TProps>,
+) => React.ComponentType<Omit<TProps, TPropName>>;
+
+export function componentAutoBinder<
+	const TPropName extends string,
+	const TValue,
```

Comment:
This syntax with `const` in the angle brackets is new to me.

i got a pretty good explanation of it just now from ChatGPT:

https://chatgpt.com/share/dd39a7d6-0fbe-4632-ba26-bf554f3448e0

Reply:
That looks like a pretty good explanation of it to me! 

> This can be particularly useful when you want to ensure that the type parameters are not widened to more general types and are kept as specific as possible... For example, if TPropName is provided as 'foo', without const, TypeScript might widen it to string, but with const, it stays as 'foo'.

That's exactly the point here; it'll ensure the property name is stored in the type parameter for type safety.